src/history.rs

```diff
@@ -48,56 +49,65 @@ const_assert_eq!(4, align_of::<ModalityProbeInstant>());
         + size_of::<LogicalClock>()
-        + size_of::<LogEntryRingBuffer<'_>>()
-        + size_of::<u32>()
-        + size_of::<u64>()
-        + size_of::<RestartSequenceCounter<'_>>(),
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<RestartSequenceCounter<'_>>()
+        + size_of::<u64>(),
     size_of::<DynamicHistory>()
 );
 
-// On 64-bit, size is rounded up to the nearest multiple of its alignment:
-// size of the fields is 116 bytes, which rounds up to
-// the next 8-byte alignment at 120 bytes (a difference of 4 bytes)
-#[cfg(not(target_pointer_width = "32"))]
+// 4 bytes of padding required before clocks list to get FixedSliceVec to 8 byte align
+#[cfg(target_pointer_width = "64")]
 const_assert_eq!(
-    4 + size_of::<ProbeId>()
+    size_of::<u32>()
+        + size_of::<ProbeId>()
+        + size_of::<LogBuffer<'_>>()
         + size_of::<u32>()
-        + size_of::<FixedSliceVec<'_, LogicalClock>>()
         + size_of::<LogicalClock>()
-        + size_of::<LogEntryRingBuffer<'_>>()
-        + size_of::<u32>()
-        + size_of::<u64>()
-        + size_of::<RestartSequenceCounter<'_>>(),
+        + 4
+        + size_of::<FixedSliceVec<'_, LogicalClock>>()
+        + size_of::<RestartSequenceCounter<'_>>()
+        + size_of::<u64>(),
     size_of::<DynamicHistory>()
 );
 
 /// Manages the core of a probe in-memory implementation
 /// backed by runtime-sized arrays of current logical clocks
 /// and probe log items
+///
+/// Note: overwrite_priority, probe_id, and log must be accessed by the debug collector using direct memory access,
+/// so they must be public fields (in order for the debug collector to calculate their offsets).
+/// No non-repr(C) fields or usizes (including refs or slices) should be put before those fields,
+/// as that would result in the offsets of those fields to change depending on the architecture of the target.
 #[derive(Debug)]
+#[repr(C)]
 pub struct DynamicHistory<'a> {
-    pub(crate) probe_id: ProbeId,
+    /// Minimum priority level of items that can be written to the log
+    pub overwrite_priority: u32,
+    /// ID of this probe
+    pub probe_id: ProbeId,
+    /// Log used to store events and trace clocks
+    pub log: LogBuffer<'a>,
     /// The number of events seen since the current
     /// probe's logical clock last increased.
     pub(crate) event_count: u32,
+    pub(crate) self_clock: LogicalClock,
     /// Invariants:
     ///   * The first clock is always that of the local probe id
     pub(crate) clocks: FixedSliceVec<'a, LogicalClock>,
-    pub(crate) self_clock: LogicalClock,
-    pub(crate) log: LogEntryRingBuffer<'a>,
-    pub(crate) log_items_missed: u32,
-    pub(crate) report_seq_num: u64,
     pub(crate) restart_counter: RestartSequenceCounter<'a>,
+    pub(crate) report_seq_num: u64,
 }
 
 #[derive(Debug)]
 struct ClocksFullError;
 
 impl<'a> DynamicHistory<'a> {
+    /// Create new DynamicHistory at given destination
```

Comment:
Technically, it'll be put somewhere inside the destination, but it's not certain that the `DynamicHistory` structure will start exactly at the beginning of the provided destination byte array, thanks to alignment-checking done in `fixed_slice_ve::single::embed`.